dist/s_clang-format

```diff
@@ -47,20 +47,34 @@ if test $? -ne 0; then
 	download_clang_format
 fi
 
+exclude=true
 case $# in
 0)
-	# Get all source files that aren't in s_clang-format.list.
-	search=`find bench examples ext src test -name '*.[ch]' -o -name '*.cxx' -o -name '*.cpp'`
-	for f in `cat dist/s_clang-format.list`; do
-		search=`echo "$search" | sed "\#$f#d"`
-	done;;
+	# Get all source files that are modified according to git.
```

Comment:
I do have some overall hesitation about doing a diff against develop branch in the default case.  It does introduce a dependency on the "way things are" git-wise, that wasn't there before (what if the WT is copied to another place where there is no "develop" branch, or if we have a subproject that has a different git structure, or we change the name of "develop"...?).

One thing that would help is having in one or more of the evergreen tasks (including PR) calling `s_all -a` (where `-a` is passed down to any subscripts where it's relevant).

Reply:
> It does introduce a dependency on the "way things are" git-wise.

Good point, @ddanderson. Maybe treat the `git diff` path as an optimization -- if we're in an environment where it works, we win. But if not, the keep the current check everything behavior as a default? I.e., have error checking around the git stuff so that we can handle the case where git isn't available, doesn't understand `develop`, etc.

Reply:
`s_fast` works for me.  I like that existing evergreen scripts work as is, and always do the "complete" check.

Reply:
@ajmorton has suggested s_fast just calls s_all with a "fast" flag which I agree with so I'll implement that change next.